tripal_chado/includes/TripalImporter/OBOImporter.inc

```diff
@@ -2553,6 +2604,14 @@ class OBOImporter extends TripalImporter {
         $response = drupal_json_decode($response->data);
       }
     }
+    elseif ($found_iri) {
+      $full_url = 'http://www.ebi.ac.uk/ols/api/ontologies/' . $ontology . '/' . $type . '/' . $found_iri;
```

Comment:
I think this may be a problem as `$type` isn't defined in this function....  

Also, maybe I confused things a bit.  The $found_iri variable?  An IRI would a full path to an object (e.g. a full URL) so the variable name confuses me here. What is in the $found_uri variable?

Reply:
in line 1049 I run a query to find the term `$ontology_results =  $this->oboEbiLookup($id, 'query');`
then there are checks. In line 1072 I go through the query results with a foreach `foreach ($ontology_results['response']['docs'] as $each ) {`. I look at each result to see if it is what we want, it should be defining_ontology=true, if not, it should have the same ID. 

```
      foreach ($ontology_results['response']['docs'] as $each ) {
        $obo_id = $each['obo_id'];
        $defining_ontology = $each['is_defining_ontology'];
        // First result should have defining_ontology=true, but if
        // it doesn't, use the first result with obo_id=$id
        if ($defining_ontology == 'false' and $obo_id != $id) {
          continue;
        }
        $iri = urlencode(urlencode($each['iri']));
        $ontologyID = $each['ontology_name'];
        // Type should be 'property' or 'class' in the response
        $type = $each['type'];
      }
    }
```
Then if it is a good hit in our query results, collect the iri, I call it $found_iri, so that it is more clear that it was retrieved from the query and not built using the accession and an ontology prefix. Since sometimes (though not too often) the iri cannot be formed some cleanly.

Does that make more sense? I query the accession, get hits, pick a good hit, get the good hit iri, then use the found iri in the term api call.

Reply:
Yes, but does the $found_iri contain a full URL?   

Also, I think the $found_iri variable should have a NULL default in the function arguments so that other function calls that are made that don't break.

Reply:
Also. I altered the oboEbiLookup to take the found_iri since it could differ from a built iri. If the found_iri is defined then it means we should use that iri in the api call.

Reply:
Ok. $type was a leftover from when this code was in the main. I addressed this now with checking for type and formatting the correct text for the api url.

Reply:
$found_iri will contain the values found in the 'iri' key: `"iri": "http://purl.obolibrary.org/obo/RO_0002206",` which is  a full iri.

See this sample api search:
https://www.ebi.ac.uk/ols/api/search?q=RO:0002206&queryFields=obo_id

Reply:
I am setting the default of $found_iri = NULL in the  oboEbiLookup function. is this the right syntax? `private function oboEbiLookup($accession, $type_of_search, $found_iri = NULL) {` Is it null or NULL?

Reply:
Yes that's right.  I'll go ahead and test.